pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -85,12 +85,18 @@
     // This saves the creation of a clone everytime, and the memory associated to all these clones.
     Field tzField;
     try {
-      tzField = TimeZone.class.getDeclaredField("defaultTimeZone");
-      tzField.setAccessible(true);
-      TimeZone defaultTz = TimeZone.getDefault();
-      Object tzFromField = tzField.get(null);
-      if (defaultTz == null || !defaultTz.equals(tzFromField)) {
-        tzField = null;
+      String javaVersion = System.getProperty("java.version");
+      tzField = null;
+      // Avoid reflective access in Java 9+
+      if (javaVersion.startsWith("1.") && javaVersion.length() == 3
+          && javaVersion.charAt(2) >= '4' && javaVersion.charAt(2) <= 8) {
```

Comment:
it should be `chatAt(2) >= '6'` and you forgot the quotes for `'8'`.

Reply:
Thanks, makes sense